fecgov/openFEC
# Fix elasticsearch backup logic

## New changes:

- Put elasticsearch backups in `es-backups` s3 directory, don't clear with nightly task
- Add slack error message for elasticsearch backup task

## Summary (required)

- Resolves issue found while testing elasticsearch backups (authentication failed upon backup creation post-deploy)
>raised unexpected: TransportError(500, 'repository_verification_exception', {'error': {'root_cause': [{'type': 'repository_verification_exception', 'reason': '[legal_s3_repository] path  is not accessible on master node'}], 'reason': 'The AWS Access Key Id you provided does not exist in our records. 
- Configure backup repository at backup and restore (needed when credentials are updated upon deploy)
- Restore both `docs` and `archived_murs` indices
- Add support for zero-downtime restore (thanks @qqss88 for the suggestion!)

## How to test the changes locally

- I tested this with a manual deploy: `invoke deploy --space dev --skip-migrations`
- See instructions for testing: https://github.com/fecgov/openFEC/wiki/Elasticsearch-backup-instructions

## Impacted areas of the application
List general components of the application that this PR will affect:

-  Elasticsearch backup and restore



## Related PRs

https://github.com/fecgov/openFEC/pull/3491: Automate elasticsearch backups
